internal/icinga2/client_api.go

```diff
@@ -306,17 +318,38 @@ func (client *Client) checkMissedChanges(ctx context.Context, objType string, ca
 		var fakeEv *event.Event
 		if checkableIsMuted && attrs.Acknowledgement != AcknowledgementNone {
 			ackComment, err := client.fetchAcknowledgementComment(ctx, hostName, serviceName, attrs.AcknowledgementLastChange.Time())
-			if err != nil {
-				return fmt.Errorf("fetching acknowledgement comment for %q failed, %w", objectName, err)
-			}
+			if errors.Is(err, errMissingAcknowledgementComment) {
+				// Unfortunately, there is no Acknowledgement object in Icinga 2, but only related runtime attributes
+				// attached to Host or Service objects. Those attributes contain no authorship. The only way to link an
+				// acknowledgement to a contact, when being fetched through the Config Objects API, is to find a
+				// matching Comment object, which contains an author field.
+				//
+				// This is not the case for the Event Stream API, where AcknowledgementSet has an author field.
+				//
+				// However, when no author is present, the Acknowledgement Event cannot be processed. Eventually, the
+				// Incident.processAcknowledgementEvent method will fail hard.
+
+				client.Logger.Infow("Cannot find the comment for an acknowledgement, creating a generic muted event",
+					zap.String("object", objectName), zap.Error(err))
```

Comment:
It doesn't make sense to include the `zap.Error()` field in an info level, and since it doesn't contain any special information other than the same set of info being logged there, I would just drop it.

Reply:
It actually contains a bit more information, indicating if there is no acknowledgment comment at all or none in the arbitrary chosen time frame of a second. As the latter feels like something were some day a bug might lurk, I would like to keep this information included.

I have changed the field type from an `Error` to a `NamedError`, naming it "reason" so no error appears on info level.

Reply:
> It actually contains a bit more information, indicating if there is no acknowledgment comment at all or none in the arbitrary chosen time frame of a second. As the latter feels like something were some day a bug might lurk, I would like to keep this information included.

The now updated log messages look fine to me, just FYI: that might just tell you if an acknowledgement with a persistent comment was created some time in the past (at least, that's how you can easily make it log the other message)